packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -1951,46 +1951,31 @@ function commitMutationEffectsOnFiber(
       commitReconciliationEffects(finishedWork);
 
       if (flags & Update) {
-        try {
-          commitHookEffectListUnmount(
-            HookInsertion | HookHasEffect,
-            finishedWork,
-            finishedWork.return,
-          );
-          commitHookEffectListMount(
-            HookInsertion | HookHasEffect,
-            finishedWork,
-          );
-        } catch (error) {
```

Comment:
This used to be one catch but is now two. So if the unmount throws, we continue with the mount.

That's probably not desirable but consistent with other effects we have atm.